components/eam/src/chemistry/modal_aero/modal_aero_convproc.F90

```diff
@@ -2168,11 +2248,10 @@ subroutine compute_fluxes_tendencies(                        &
                          - ( md_i(kp1)*cond(icnst,kp1)                          &
                            + md_i(kk )*min(chat(icnst,kk ),const(icnst,kk )) )
             else
-               fluxin  =     mu_i(kp1)*conu(icnst,kp1) - ( md_i(kk )*cond(icnst,kk ) )
-               fluxout =     mu_i(kk )*conu(icnst,kk ) - ( md_i(kp1)*cond(icnst,kp1) )
-
                ! new method -- simple upstream method for the env subsidence
                ! tmpa = net env mass flux (positive up) at top of layer k
+               fluxin  =     mu_i(kp1)*conu(icnst,kp1) - ( md_i(kk )*cond(icnst,kk ) )
+               fluxout =     mu_i(kk )*conu(icnst,kk ) - ( md_i(kp1)*cond(icnst,kp1) )
                tmpa = -( mu_i(kk  ) + md_i(kk  ) )
                if (tmpa <= 0.0_r8) then
```

Comment:
There are some `0.0` values without `_r8` at the end lurking around. Do we care about these? Zero is a "special" floating point number in that it has an exact representation, so we could be more lenient with this particular value.

Reply:
as commented by Balwinder, I will keep the numbers as it as the original code